packages/git/src/browser/blame/blame-decorator.ts

```diff
@@ -0,0 +1,203 @@
+                style.content = `'${content}'`;
+                style.borderColor = heat;
+            }));
+            const hoverMessage = this.formatHover(commit);
+            hoverMessages.set(sha, hoverMessage);
+        }
+        const commitLines = blame.lines;
+        const highlightedCommitLine = commitLines.find(c => c.line === highlightLine);
+        const highlightedSha = highlightedCommitLine ? highlightedCommitLine.sha : '';
+        let previousLineSha = '';
+        const editorDecorations: EditorDecoration[] = [];
+
+        for (const commitLine of commitLines) {
+            const { line, sha } = commitLine;
+            const beforeContentClassName = beforeContentStyles.get(sha)!.className;
+            const hoverMessage = hoverMessages.get(sha)!;
+            const options = <EditorDecorationOptions>{
+                beforeContentClassName,
+                hoverMessage,
+            };
+            if (sha === highlightedSha) {
+                options.beforeContentClassName += ' ' + BlameDecorator.highlightStyle.className;
+            }
+            if (sha === previousLineSha) {
+                options.beforeContentClassName += ' ' + BlameDecorator.continuationStyle.className;
+            }
+            previousLineSha = sha;
+            const range = Range.create(Position.create(line, 0), Position.create(line, 0));
+            editorDecorations.push(<EditorDecoration>{ range, options });
+        }
+        const styles = [...beforeContentStyles.values()];
+        return { editorDecorations, styles };
+    }
+
+    protected formatHover(commit: Commit): string {
+        const date = new Date(commit.author.timestamp);
+        return `
+        ${commit.sha}
+
+        ${commit.author.name}, ${date.toString()}
+
+        ${commit.summary}`;
+    }
+
+    protected formatContentLine(commit: Commit, daysFromNow: number): string {
+        let when = `${daysFromNow} days ago`;
+        if (daysFromNow < 1) {
+            when = 'today';
+        } else if (daysFromNow === 2) {
+            when = 'yesterday';
+        }
+        const contentWidth = BlameDecorator.maxWidth - when.length - 2;
+        let content = commit.summary.substring(0, contentWidth + 1);
+        content.replace('\n', '↩︎');
+        if (content.length > contentWidth) {
+            let cropAt = content.lastIndexOf(' ', contentWidth - 4);
+            if (cropAt < contentWidth / 2) {
+                cropAt = contentWidth - 3;
+            }
+            content = content.substring(0, cropAt) + '...';
+        }
+        if (content.length < contentWidth) {
+            content = content + '\u2007'.repeat(contentWidth - content.length); // fill up with blanks
+        }
+        return `${content} ${when}`;
+    }
+
+    protected today = (() => {
+        const dateNow = new Date();
+        return Date.UTC(dateNow.getFullYear(), dateNow.getMonth(), dateNow.getDate());
+    })();
+
+    protected daysFromNow(date: Date): number {
```

Comment:
there is a package to display dates which is used by github: https://www.npmjs.com/package/moment

Reply:
'will use it! thanks!